packages/hub/src/scope.ts

```diff
@@ -441,9 +447,20 @@ export class Scope implements ScopeInterface {
     event.breadcrumbs = [...(event.breadcrumbs || []), ...this._breadcrumbs];
     event.breadcrumbs = event.breadcrumbs.length > 0 ? event.breadcrumbs : undefined;
 
+    event.processingMetadata = this._processingMetadata;
+
     return this._notifyEventProcessors([...getGlobalEventProcessors(), ...this._eventProcessors], event, hint);
   }
 
+  /**
+   * Add data which will be accessible during event processing but won't get sent to Sentry
```

Comment:
There shouldn't be docs about the behavior of other places: hard to maintain and easy to have it wrong with new changes.
```suggestion
   * Add data which will be accessible during event processing
```

Reply:
That's the entire point of this field, though - it's the scratch pad, where you can stick anything you want without worrying about having to pluck it (and only it) out, maybe deleting the whole container, etc. We just delete this entire field before the event is sent, and we know we haven't accidentally deleted too much or too little. (That's why it's called processingMetadata - it's data that's only useful for the SDK's event processing pipeline.

Reply:
Disagree - otherwise, how does someone reading the code know it's safe to stick stuff there, without worrying about it getting sent to Sentry? Its entire raison d'etre is to be a local (and local only) datastore, which doesn't actually turn into anything. Kinda like `os.tmpdir()` in node.

The fact that we're even having this conversation makes me wonder if there's a better name for it, though. Can you think of a name which would make it clearer what it's there for? `localProcessingMetadata`? Maybe it would help to stick a `_` at the front? WDYT?

Reply:
Renaming sounds good. The names I can think of are probably too long; for example, `sdkEventProcessingMetadata`. I rather `sdk` over `local` since IMO it's more accurate and easier to understand. I don't like `eventProcessingMetadata` since it doesn't reference it happening in the SDK. I've been some time thinking about this and can't get a good name (good also including short), so choose what you think fits best. I'm only opinionated against adding `_` to the prefix, it's easy to relate it to a private class attribute.

Reply:
Another thing to consider is that the name must be very clear and should not be mixed with `debug_meta`.

Reply:
I went with `sdkProcessingMetadata`. Slightly shorter than `sdkEventProcessingMetadata`, but hopefully specific enough to get the point across.